File: tests-playwright/src/keywords/api-explorer.keywords.ts
import { test } from '../fixtures/fixtures'
import type { ApiExplorerPage } from '../pages/api-explorer/api-explorer-page'

export async function action_openApiExplorer(apiExplorerPage: ApiExplorerPage): Promise<void> {
  await test.step('Open the API explorer page', async () => {
    await apiExplorerPage.open()
  })
}

export async function check_apiExplorerShellReady(apiExplorerPage: ApiExplorerPage): Promise<void> {
  await test.step('Verify the request groups and empty response state render', async () => {
    await apiExplorerPage.verifyShellReady()
  })
}

File: tests-selenium/src/test/java/com/coinpulse/selenium/keywords/ApiExplorerKeywords.java
package com.coinpulse.selenium.keywords;

import com.coinpulse.selenium.pages.ApiExplorerPage;
import io.qameta.allure.Allure;
import org.openqa.selenium.WebDriver;

/**
 * API Explorer keywords: specs call these, never the page object directly.
 */
public class ApiExplorerKeywords {

    private final ApiExplorerPage apiExplorerPage;

    public ApiExplorerKeywords(WebDriver driver) {
        this.apiExplorerPage = new ApiExplorerPage(driver);
    }

    public void actionOpenApiExplorer() {
        Allure.step("Open the API Explorer page", apiExplorerPage::open);
    }

    public void checkApiExplorerShellReady() {
        Allure.step("Verify the API Explorer renders its request groups and empty response state",
                apiExplorerPage::verifyShellReady);
    }
}
